### PR TITLE
network: add Pprof metrics

### PR DESCRIPTION
**Pull request by volekerb**
Since we have some perf issues from time to time it is good to have pprof debugger

Usage: 
addr (host + port) + 
```
	"/debug/pprof/"
	"/debug/pprof/cmdline"
	"/debug/pprof/profile"
	"/debug/pprof/symbol"
	"/debug/pprof/trace"
```
Link for documentation: https://golang.org/pkg/net/http/pprof/

### CODE REVIEW

**Comment by roman-khimov** [comment]
Yet another http instance in the node is a little frightening to me, makes me think about possibilities to make it conditional to build. If we're to keep it non-conditional than the only thing I probably have on it the default settings, I don't think it's good to enable it by default for mainnet and tesnet. Frankly, even for privnet it's debatable, because that's a node developer tool, it gives nothing the end user.

**Comment by volekerb** [comment]
Disabled by default, but we can enable on our testing nodes.

File: pkg/network/metrics/pprof.go [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+	"net/http"
+	"net/http/pprof"
+)
+// PprofService https://golang.org/pkg/net/http/pprof/.
+type PprofService Service
+
+// NewPprofService created new service for gathering pprof metrics.
+func NewPprofService(cfg Config) *Service {
+	handler := http.NewServeMux()
+	handler.HandleFunc("/debug/pprof/", pprof.Index)
```
Comment (roman-khimov) [inline pkg/network/metrics/pprof.go]: BTW, do we really need to push it down here if that's the only service bound to this instance of `http`? Maybe move it to the `/` and be done with it?

**Comment by roman-khimov** [comment]
And maybe it would also be useful to add some paragraph mentioning this feature somewhere, otherwise it's only visible in the config and this PR.

File: config/config.go [hunk 2]
```diff
@@ -76,7 +76,8 @@ type (
 		MaxPeers          int                      `yaml:"MaxPeers"`
 		AttemptConnPeers  int                      `yaml:"AttemptConnPeers"`
 		MinPeers          int                      `yaml:"MinPeers"`
-		Monitoring        metrics.PrometheusConfig `yaml:"Monitoring"`
+		Monitoring        metrics.Config           `yaml:"Monitoring"`
```
Comment (roman-khimov) [inline config/config.go]: It just occured to me that now this name is probably too ambiguos, even in README it's not obvious what kind of monitoring we're configuring here. Maybe rename it to `Prometheus` now?

File: README.md [hunk 3]
```diff
@@ -85,6 +85,19 @@ Available network flags:
...
 - `--testnet, -t`
 
+#Developer notes
+Nodes have such features as [Prometheus](https://prometheus.io/docs/guides/go-application) and 
+[Pprof](https://golang.org/pkg/net/http/pprof/) in order to have additional information about them for debugging.
+
+How to configure Prometheus or Pprof:
+In `config/protocol.*.yml` there is 
+```
+  Monitoring:
```
Comment (roman-khimov) [inline README.md]: Now this one is outdated a little.

File: config/protocol.testnet.yml [hunk 4]
```diff
@@ -60,3 +60,6 @@ ApplicationConfiguration:
   Monitoring:
```
Comment (roman-khimov) [inline config/protocol.testnet.yml]: And these are too, BTW.

File: README.md (hunk 3 above)
Comment (volekerb) [reply README.md]: pow wow true

File: config/protocol.testnet.yml (hunk 4 above)
Comment (volekerb) [reply config/protocol.testnet.yml]: good catch!

**Comment by roman-khimov** [comment]
ACK, but for 0.71.0.

File: cli/server/server.go [hunk 5]
```diff
@@ -283,12 +283,14 @@ func startServer(ctx *cli.Context) error {
 	server := network.NewServer(serverConfig, chain)
 	rpcServer := rpc.NewServer(chain, cfg.ApplicationConfiguration.RPC, server)
 	errChan := make(chan error)
-	monitoring := metrics.NewMetricsService(cfg.ApplicationConfiguration.Monitoring)
+	prometheus := metrics.NewPrometheusService(cfg.ApplicationConfiguration.Prometheus)
+	pprof := metrics.NewPprofService(cfg.ApplicationConfiguration.Pprof)
```
Comment (fyrchik) [inline cli/server/server.go]: Is also helpful to have metrics of other commands (at least of `db restore`).

Comment (volekerb) [reply cli/server/server.go]: `db dump` is the case as well, right?
And what kind of metrics are interesting? All?

Comment (volekerb) [reply cli/server/server.go]: Figures out: db dump and restore could have pprof and it will help us.

Comment (roman-khimov) [reply cli/server/server.go]: Time to update and merge this one.

Comment (volekerb) [reply cli/server/server.go]: added metrics @fyrchik @roman-khimov

File: cli/server/server.go [hunk 6]
```diff
@@ -183,8 +198,11 @@ func dumpDB(ctx *cli.Context) error {
 		}
 	}
 	chain.Close()
+	prometheus.ShutDown()
+	pprof.ShutDown()
```
Comment (roman-khimov) [inline cli/server/server.go]: There has to be some symmetry, I think. If we start A, B and C then we should stop C, B and A.

Comment (volekerb) [reply cli/server/server.go]: fixed!

**Comment by volekerb** [comment]
@fyrchik fixed paths, thanks for catching!

**Pull request merged**